doc/source/area-detector.rst

```diff
@@ -269,6 +269,194 @@ The vendor specific details are embedded in the cams
+       optional trigger_mixin class and can be swapped out for any other
+       class that inherits from
+       :class:`ophyd.areadetector.trigger_mixins.TriggerBase`.  These
+       classes provide the logic to 'trigger' the detector and actually
+       acquire the images.
+
+    3. PluginClass can be
+       :class:`ophyd.areadetector.plugins.PluginBase`,
+       :class:`ophyd.areadetector.cam.CamBase` or any plugin/cam class
+       that inherits from either of these.
+
+    4. In the ophyd source code, you may see
+       :class:`.ophyd.areadetector.base.ADComponent`
+       used. Functionally, this is interchangeable with an ordinary
+       :class:`.ophyd.device.Component` (imported as ``Cpt`` above); it
+       just adds extra machinery for generating a docstring based on a
+       scrape of the HTML of the official AreaDetector documentation. For
+       custom extensions such as we are addressing here, it is not
+       generally applicable.
+
+
+The Areadetector device should then be instantiated using:
+
+.. code-block:: python
+
+    ADdevice_name = Some_Areadetector_Device_Class(Areadetector_device_PV_suffix,
+                                                   name = 'ADdevice_name')
+
+
+Custom Plugins or Cameras
+=========================
+
+For custom hardware based on area-detector it may be necesary to add a
+custom plugin or camera class, this section will cover what is
+required. Both 'plugins' and 'cameras' act in the same way, but have
+slightly different 'base' attributes, hence they have different 'base
+classes'.  New Plugin classes should inherit from
+:class:`ophyd.areadetector.base.PluginBase` while new Camera classes
+should inherit from :class:`ophyd.areadetector.cam.CamBase`.  Both
+should have the following PV structure (replace 'plugin' with 'cam'
+for cameras):
+
+.. code-block:: python
+
+    PV = 'Areadetector_device_PV_prefix:Plugin_suffix:attribute_suffix'
+
+As an example, for the 'max value' component of the built-in areadetector
+'stats' class this looks like:
+
+.. code-block:: python
+
+    PV = 'Areadetector_device_PV_prefix:Stats:max_value'
+
+where ``Areadetector_device_PV_prefix`` is the PV name for the Area
+detector device, ``plugin_suffix = Stats`` is the 'stats' Plugin
+suffix and ``attribute_suffix = max_value`` is the 'max value'
+attribute suffix.
+
+
+In order to create the class then the following code is required
+(where ``XXX`` is the name of the plugin:
+
+.. code-block:: python
+
+    from ophyd.areadetector.base import ad_group, EpicsSignalWithRBV
+    from ophyd.signal import EpicsSignal, EpicsSignalRO
+    from ophyd.device import DynamicDeviceComponent as DDCpt, Component as Cpt
+    from ophyd.areadetector.plugins import PluginBase
+    from ophyd.areadetector.filestore_mixins import FileStoreHDF5
+
+    Class XXXplugin(PluginBase, FileStoreHDF5):
+        '''An areadetector plugin class that does ......'''
+        _suffix_re = 'Plugin_suffix\d:'
```

Comment:
Is it used anywhere?

Reply:
https://github.com/NSLS-II/ophyd/blob/183c09c70ce463a4f46fe6562eb458d88c1b0dae/ophyd/areadetector/plugins.py#L942-L953

Probably should mention `register_plugin` too?  I am content to let that slide though...